scripts/ci/jobs/diff-dumps.sh

```diff
@@ -107,48 +107,88 @@ create_offline_dump() {
+
+    local curl="curl --silent --show-error --max-time 60 --retry 3"
+    local release_version_url="https://raw.githubusercontent.com/stackrox/stackrox/master/scanner/updater/version/RELEASE_VERSION"
+    local vuln_version_pattern="https://raw.githubusercontent.com/stackrox/stackrox/%s/scanner/VULNERABILITY_VERSION"
+
+    [ -f scanner-defs.zip ] || die "missing required bundle inclusion: scanner-defs.zip"
+    [ -f k8s-istio.zip ]    || die "missing required bundle inclusion: k8s-istio.zip"
+    [ -f "$latest_bundle" ] || die "missing latest bundle: $latest_bundle"
+
+    info "existing latest bundle:"
+    unzip -l "$latest_bundle"
+
+    info "fetching release versions at $release_version_url"
+
+    # Read the supported releases file, filter lines with release versions,
+    # parse 4.4.* and 4.5.* to print only the X.Y, otherwise print the whole
+    # X.Y.Z, finally merge everything together, and pipe one release per line.
+
+    $curl --fail "$release_version_url" \
+        | grep '^[0-9]\.[0-9]\.[0-9]$' \
+        | sed -En 's/^(4\.4|4\.5)\..*/\1/p;t; p' \
+        | sort -V \
+        | uniq \
+        | while read -r release _; do
+              case "$release" in
+                  4.4)
+                      # We don't support 4.4 offline bundles.
+                      continue
+                      ;;
+                  4.5)
+                      # Backward compatibility, the "schema version" is the release itself.
+                      version="$release"
+                      ;;
+                  *)
+                      info "fetching schema version for release $release"
+                      tmp=$(mktemp)
+                      vuln_version_url=$(printf "$vuln_version_pattern" "$release")
+                      case $($curl --write-out "%{http_code}" -o "$tmp" "$vuln_version_url") in
+                          200)
+                              version=$(cat $tmp)
+                              ;;
+                          404)
+                              info "release tag not found, assuming the release was not cut, skipping..."
+                              continue
+                              ;;
+                          *)
+                              die "failed to fetch the v4 offline bundle version for $release: status $status"
+                              ;;
+                      esac
+                      ;;
+              esac
+
+              local v4_bundle="scanner-v4-defs-$version.zip"
+              info "building $release using schema version $version: filename: $v4_bundle"
+
+              $curl \
+                  --fail \
+                  -o "$v4_bundle" \
+                  "https://definitions.stackrox.io/v4/offline-bundles/$v4_bundle"
+              zip "$latest_bundle" "$v4_bundle"
+              zip "scanner-vulns-$release.zip" scanner-defs.zip k8s-istio.zip "$v4_bundle"
+          done
+}
 
+scanner_v4_upload_bundles() {
```

Comment:
nit: 
```suggestion
scanner_upload_bundles() {
```
or revert it back to the old name, `upload_offline_dumps`, since the upload is not V4 specific.

Reply:
92a330e0